### PR TITLE
docs(style): [result] use new display tag

### PR DESCRIPTION
**Pull request by wzc520pyfm**
Related: #10754 

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/result/src/result.vue [hunk 1]
```diff
@@ -35,7 +35,7 @@ defineOptions({
 
 const props = defineProps(resultProps)
 
-const ns = useNamespace('result')
+const nsResult = useNamespace('result')
```
Comment (chenxch) [inline packages/components/result/src/result.vue:38]: There is no need to change the variable name.

File: docs/en-US/component/result.md [hunk 2]
```diff
@@ -23,19 +23,21 @@ result/customized-content
...
-| sub-title | custom sub title  |
-| extra     | custom extra area |
+## API
+
+### Attributes
+
+| Name      | Description         | Type                                                 | Default |
+| --------- | ------------------- | ---------------------------------------------------- | ------- |
+| title     | title of result     | ^[string]                                            | —       |
+| sub-title | sub title of result | ^[string]                                            | —       |
```
Comment (chenxch) [inline docs/en-US/component/result.md:33]: default `''`

File: docs/en-US/component/result.md [hunk 3]
```diff
@@ -23,19 +23,21 @@ result/customized-content
...
-| title     | custom title      |
-| sub-title | custom sub title  |
-| extra     | custom extra area |
+## API
+
+### Attributes
+
+| Name      | Description         | Type                                                 | Default |
+| --------- | ------------------- | ---------------------------------------------------- | ------- |
+| title     | title of result     | ^[string]                                            | —       |
```
Comment (chenxch) [inline docs/en-US/component/result.md:32]: default `''`

File: docs/en-US/component/result.md (hunk 2 above)
Comment (wzc520pyfm) [reply docs/en-US/component/result.md]: updated

File: packages/components/result/src/result.vue (hunk 1 above)
Comment (wzc520pyfm) [reply packages/components/result/src/result.vue]: updated

Comment (wzc520pyfm) [reply packages/components/result/src/result.vue]: updated

**Review by chenxch — APPROVED** [review]

**Comment by chenxch** [comment]
Good job.

**Review by ryuhangyeong — APPROVED** [review]

**Review by btea — APPROVED** [review]

**Pull request merged**